kratos/geometries/triangle_3d_3.h

```diff
@@ -1739,6 +1739,63 @@ template<class TPointType> class Triangle3D3
         return rResult;
     }
 
+    ///@}
+    ///@name Spatial Operations
+    ///@{
+
+    /**
+    * @brief Projects a certain point on the geometry, or finds
+    *        the closest point, depending on the provided
+    *        initial guess. The external point does not necessary
+    *        lay on the geometry.
+    *        It shall deal as the interface to the mathematical
+    *        projection function e.g. the Newton-Raphson.
+    *        Thus, the breaking criteria does not necessarily mean
+    *        that it found a point on the surface, if it is really
+    *        the closest if or not. It shows only if the breaking
+    *        criteria, defined by the tolerance is reached.
+    *
+    *        This function requires an initial guess, provided by
+    *        rProjectedPointLocalCoordinates.
+    *        This function can be a very costly operation.
+    *
+    * @param rPointGlobalCoordinates the point to which the
+    *        projection has to be found.
+    * @param rProjectedPointGlobalCoordinates the location of the
+    *        projection in global coordinates.
+    * @param rProjectedPointLocalCoordinates the location of the
+    *        projection in local coordinates.
+    *        The variable is as initial guess!
+    * @param Tolerance accepted of orthogonal error to projection.
+    * @return It is chosen to take an int as output parameter to
+    *         keep more possibilities within the interface.
+    *         0 -> failed
+    *         1 -> converged
+    */
+    int ProjectionPoint(
+        const CoordinatesArrayType& rPointGlobalCoordinates,
+        CoordinatesArrayType& rProjectedPointGlobalCoordinates,
+        CoordinatesArrayType& rProjectedPointLocalCoordinates,
+        const double Tolerance = std::numeric_limits<double>::epsilon()
+        ) const override
+    {
+        // We compute the normal to check the normal distances between the point and the triangles, so we can discard that is on the triangle
+        const auto center = this->Center();
+        const array_1d<double, 3> normal = this->UnitNormal(center);
+
+        // We compute the distance, if it is not in the plane we project
+        const Point point_to_project(rPointGlobalCoordinates);
+        double distance;
+        rProjectedPointGlobalCoordinates = GeometricalProjectionUtilities::FastProject( center, point_to_project, normal, distance);
+
+        // Projecting
+        noalias(rProjectedPointGlobalCoordinates) = rPointGlobalCoordinates - normal * distance;
+
+        PointLocalCoordinates( rProjectedPointLocalCoordinates, rProjectedPointGlobalCoordinates );
+
+        return 1;

```

Comment:
I don't understand why you are calculating the distance which is a costly operation by itself. I would do something like:

```c++
        PointLocalCoordinates(rProjectedPointLocalCoordinates, rPointGlobalCoordinates);

        for (std::size_t  i = 0; i < 3; i++) {
            rProjectedPointLocalCoordinates[i] = (rProjectedPointLocalCoordinates[i] < 0.0) ? 0.00 :  rProjectedPointLocalCoordinates[i];
            rProjectedPointLocalCoordinates[i] = (rProjectedPointLocalCoordinates[i] > 1.0) ? 1.00 :  rProjectedPointLocalCoordinates[i];
        }

        TheTriangle.GlobalCoordinates(rProjectedPointGlobalCoordinates, rProjectedPointLocalCoordinates);

```

Reply:
I will try, I don't know if the local coordinates do a projection

Reply:
It works!!

Reply:
wait
why does `PointLocalCoordinates` do a projection? 😅 

now the code is onconsistent

Line and Quad do the projection (in `ProjectionPoint`), the Triangle does not

sorry for still blocking but I want to clarify this before the geometry interface becomes even more messy :(

Reply:
`PointLocalCoordinates` in the triangle the local coordinates are implicitly computing the local coordinates in a parallel plane (it is assumed that way, that's what I understood from @pooyan-dadvand), being equivalent to the projection

Reply:
Hm ok
I think I am a bit too confused here

in any case I will approve, guess it is fine
Thx for the patience